src/Connection/Connection.hpp

```diff
@@ -2,11 +2,15 @@
 #define PROXYTCP_CONNECTION_HPP
 
 #include <memory>
+#include <csignal>
+#include "sys/socket.h"
 #include "ByteStream/ByteStream.hpp"
 
+
 namespace Proxy
 {
     class ConversationPipeline;
+    class Status;
```

Comment:
Why do you use forward declaration for Status here?
U'd like to include Status.hpp